cli/build.gradle

```
@@ -1,15 +1,21 @@
 plugins {
-    id 'bio.terra.pfb.java-common-conventions'
+    id 'bio.terra.pfb.java-library-conventions'
+    id 'maven-publish'
+    id 'com.jfrog.artifactory' version '5.0.3'
+    id 'com.gorylenko.gradle-git-properties' version '2.3.1'
 }
 
 version 'unspecified'
+apply from: "publishing.gradle"
+apply from: "generators.gradle"
 
 repositories {
     mavenCentral()
 }
 
 dependencies {
     implementation 'info.picocli:picocli:4.7.4'
+    implementation 'bio.terra:java-pfb-library:0.1.0-SNAPSHOT'
```

This is probably not what you want to do. I think what you want here is to depend on the version of the library subproject in the repo, vs a version that you're fetching from artifactory. The way to do that is:
```suggestion
    implementation project(':library')
```

---

That's much better, thank you! I was copying how we do this in the data repo client. I suppose we would want to actually import the maven artifact if we wanted to release the CLI separately from the library, right? But, the CLI is just a wrapper around the library, so I don't see why we'd run into that case.

---

> I suppose we would want to actually import the maven artifact if we wanted to release the CLI separately from the library, right?

It depends on if you want to release a version of the cli that uses a different version of the library. I think it's better to release the two in sync, so the cli and library version are always the same. Otherwise you'd need two different version numbers to track each object separately.

The data repo setup isn't great and it'd be nice to update it at some point to use subprojects like this rather than having a separate gradle project for the client tests.

---

> I think it's better to release the two in sync, so the cli and library version are always the same.
Agreed!